### PR TITLE
Adds additional validation for sending based methods

### LINKED ISSUE #80: Properly Sanitize Inputs Used for Transfer
A couple of functions require quantity input:

- [ ] sendJUP()
- [ ] sendAsset()
- [ ] placeOrder()

The inputs used for these help guide the user to entering the appropriate amounts, but since we're dealing with a financial product the functions facilitating the actual send need to do their own validation to reduce the potential for errors. 

Review comment: https://github.com/Up-North-Coding/jovian/pull/76#discussion_r938119984

### PR DESCRIPTION
**Pull request by outragedhuman**
Fixes #80

Adds an additional layer of validation before allowing send methods (referring to these as "private" methods). 

Created for future enhancement: https://github.com/Up-North-Coding/jovian/issues/97

### CODE REVIEW

**Review by outragedhuman — COMMENTED** [review]
self review done

File: src/utils/validation/index.ts [hunk 1]
```diff
@@ -42,3 +42,35 @@ export function isValidQuantity(quantity: string) {
...
+  }
+
+  // cannot be negative
+  if (parseInt(value) < 0) {
+    console.error("value cannot be negative");
+    return false;
+  }
+
+  // maximum possible value for JUP should be total supply
+  if (parseInt(value) > 1000000000 ** PrecisionExponent) {
```
Comment (vwalsh) [inline src/utils/validation/index.ts:64]: 1000000000 seems like a magic number related to the total supply. perhaps a good idea to put this into constants?

especially if supply can ever change?

**Review by vwalsh — COMMENTED** [review]
reviewed

File: src/utils/validation/index.ts (hunk 1 above)
Comment (outragedhuman) [reply src/utils/validation/index.ts:64]: done

**Pull request merged**